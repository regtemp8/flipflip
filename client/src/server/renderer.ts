import createCache, { type EmotionCache } from '@emotion/cache'
import { configureStore } from '@reduxjs/toolkit'
import { AppStorage, SystemConstants } from 'flipflip-common'
import createAppReducer from '../store/app/slice'
import createAudioReducer from '../store/audio/slice'
import createCaptionScriptReducer from '../store/captionScript/slice'
import createClipReducer from '../store/clip/slice'
import createLibrarySourceReducer from '../store/librarySource/slice'
import createPlaylistReducer from '../store/playlist/slice'
import createSceneReducer from '../store/scene/slice'
import createSceneGroupReducer from '../store/sceneGroup/slice'
import createTagReducer from '../store/tag/slice'
import playersSlice from '../store/player/slice'
import videoClipperSlice from '../store/videoClipper/slice'
import scenePickerSlice from '../store/scenePicker/slice'
import captionScriptorSlice from '../store/captionScriptor/slice'
import sceneDetailSlice from '../store/sceneDetail/slice'
import audioLibrarySlice from '../store/audioLibrary/slice'
import sourceScraperSlice from '../store/sourceScraper/slice'
import createConstantsReducer from '../store/constants/slice'
import { fromAppStorage } from '../store/app/convert'
import createDisplayReducer from '../store/display/slice'
import createDisplayViewReducer from '../store/displayView/slice'
import createScenePlaylistItemReducer from '../store/scenePlaylistItem/slice'
import createDisplayPlaylistItemReducer from '../store/displayPlaylistItem/slice'

export function createEmotionCache(): EmotionCache {
  return createCache({ key: 'css' })
}

export function createReduxStore(
  appStorage?: AppStorage,
  constants?: SystemConstants
) {
  const state = appStorage != null ? fromAppStorage(appStorage) : undefined
  return configureStore({
    reducer: {
      app: createAppReducer(state?.app),
      audio: createAudioReducer(state?.audio),
      captionScript: createCaptionScriptReducer(state?.captionScript),
      clip: createClipReducer(state?.clip),
      librarySource: createLibrarySourceReducer(state?.librarySource),
      playlist: createPlaylistReducer(state?.playlist),
      displayPlaylistItem: createDisplayPlaylistItemReducer(
        state?.displayPlaylistItem
      ),
      scenePlaylistItem: createScenePlaylistItemReducer(
        state?.scenePlaylistItem
      ),
      scene: createSceneReducer(state?.scene),
      sceneGroup: createSceneGroupReducer(state?.sceneGroup),
      display: createDisplayReducer(state?.display),
      displayView: createDisplayViewReducer(state?.displayView),
      tag: createTagReducer(state?.tag),
      constants: createConstantsReducer(constants),
      // components
      players: playersSlice,
      videoClipper: videoClipperSlice,
      scenePicker: scenePickerSlice,
      captionScriptor: captionScriptorSlice,
      sceneDetail: sceneDetailSlice,
      audioLibrary: audioLibrarySlice,
      sourceScraper: sourceScraperSlice
    }
  })
}
