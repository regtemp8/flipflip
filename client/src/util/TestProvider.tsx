import * as React from 'react'
import { Provider } from 'react-redux'
import { createTheme } from '@mui/material'
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles'
import defaultTheme from 'flipflip-common'

export default class TestProvider extends React.Component {
  readonly props: {
    children: any
    store: any
  }

  disableTransitions = {
    defaultProps: {
      disablePortal: true,
      hideBackdrop: true,
      TransitionComponent: ({ children }) => children
    }
  }

  render() {
    const theme = createTheme({
      ...defaultTheme,
      // disable Transitions when runing tests, throws errors otherwise.
      components: {
        MuiMenu: this.disableTransitions,
        MuiPopover: this.disableTransitions,
        MuiDialog: this.disableTransitions,
        MuiModal: this.disableTransitions
      }
    } as any)
    return (
      <Provider store={this.props.store}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>
        </StyledEngineProvider>
      </Provider>
    )
  }
}
