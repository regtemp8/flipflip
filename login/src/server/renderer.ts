import createCache, { type EmotionCache } from "@emotion/cache";

export function createEmotionCache(nonce?: string): EmotionCache {
  return createCache({ key: "css", nonce });
}
