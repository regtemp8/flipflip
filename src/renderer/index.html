<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>FlipFlip</title>
    <!-- Responsive meta tag -- ensures proper rendering and touch zooming for all devices -->
    <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no" />
    <!-- Roboto Font -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
    <!-- Font Icons -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />

    <meta
      http-equiv="Content-Security-Policy"
      content="frame-src https://www.imagefap.com http://www.imagefap.com https://hypno.nimja.com/ http://hypno.nimja.com/; font-src 'self' https://fonts.gstatic.com; style-src 'self' https://fonts.googleapis.com 'unsafe-inline'"
    />
  </head>

  <body>
    <div id="app" class="app"></div>
  </body>
</html>
