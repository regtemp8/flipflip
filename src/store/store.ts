import { configureStore } from '@reduxjs/toolkit'
import appSlice from './app/slice'
import audioSlice from './audio/slice'
import constantsSlice from './constants/slice'
import playersSlice from './player/slice'
import captionScriptSlice from './captionScript/slice'
import clipSlice from './clip/slice'
import librarySourceSlice from './librarySource/slice'
import overlaySlice from './overlay/slice'
import playlistSlice from './playlist/slice'
import sceneSlice from './scene/slice'
import sceneGridSlice from './sceneGrid/slice'
import sceneGroupSlice from './sceneGroup/slice'
import tagSlice from './tag/slice'
import videoClipperSlice from './videoClipper/slice'
import scenePickerSlice from './scenePicker/slice'
import captionScriptorSlice from './captionScriptor/slice'
import sceneDetailSlice from './sceneDetail/slice'
import audioLibrarySlice from './audioLibrary/slice'
import sourceScraperSlice from './sourceScraper/slice'

const store = configureStore({
  reducer: {
    app: appSlice,
    audio: audioSlice,
    captionScript: captionScriptSlice,
    clip: clipSlice,
    constants: constantsSlice,
    librarySource: librarySourceSlice,
    overlay: overlaySlice,
    playlist: playlistSlice,
    scene: sceneSlice,
    sceneGrid: sceneGridSlice,
    sceneGroup: sceneGroupSlice,
    tag: tagSlice,
    // components
    players: playersSlice,
    videoClipper: videoClipperSlice,
    scenePicker: scenePickerSlice,
    captionScriptor: captionScriptorSlice,
    sceneDetail: sceneDetailSlice,
    audioLibrary: audioLibrarySlice,
    sourceScraper: sourceScraperSlice
  }
})

export default store
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
