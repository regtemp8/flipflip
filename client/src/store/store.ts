/// <reference path="../global.d.ts"/>
import { createReduxStore } from '../server/renderer'

const store = createReduxStore(
  window.flipflipAppStorage,
  window.flipflipConstants
)

export default store
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
