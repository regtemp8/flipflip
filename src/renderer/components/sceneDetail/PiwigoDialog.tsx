import React, { useState } from 'react'
import wretch from 'wretch'
import AbortAddon from 'wretch/addons/abort'
import FormUrlAddon from 'wretch/addons/formUrl'
import Sortable from 'react-sortablejs'

import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  FormControl,
  MenuItem,
  Select,
  type Theme,
  List,
  ListItemAvatar,
  Avatar,
  ListItemText,
  Container,
  Card,
  CardContent,
  Typography,
  Checkbox,
  FormControlLabel,
  Tooltip,
  Divider,
  Chip,
  ListItemButton,
  Rating
} from '@mui/material'
import createStyles from '@mui/styles/createStyles'
import withStyles, { type WithStyles } from '@mui/styles/withStyles'

import ArrowDownwardIcon from '@mui/icons-material/ArrowDownward'
import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward'
import ExpandMoreIcon from '@mui/icons-material/ExpandMore'
import ExpandLessIcon from '@mui/icons-material/ExpandLess'

import { AF, PW, PWS } from '../../data/const'
import { arrayMove } from '../../data/utils'
import en from '../../data/en'
import { useAppSelector } from '../../../store/hooks'
import { selectConstants } from '../../../store/constants/selectors'
import {
  selectAppConfigRemoteSettingsPiwigoHost,
  selectAppConfigRemoteSettingsPiwigoPassword,
  selectAppConfigRemoteSettingsPiwigoProtocol,
  selectAppConfigRemoteSettingsPiwigoUsername
} from '../../../store/app/selectors'

const styles = (theme: Theme) =>
  createStyles({
    list: {},
    subList: {
      paddingLeft: 10
    },
    rootInput: {
      marginLeft: theme.spacing(2),
      flexGrow: 1
    },
    sortListDisabled: {
      pointerEvents: 'none',
      opacity: 0.6
    },
    sortCol: {
      marginBottom: 5
    },
    sortColContent: {
      display: 'flex',
      padding: '5px !important'
    },
    sortColLabel: {
      lineHeight: 2,
      marginRight: 'auto'
    },
    sortColDir: {
      marginTop: 8
    },
    tagChip: {
      marginTop: 0,
      marginRight: 5,
      marginBottom: 5,
      marginLeft: 0
    },
    areaHeaderFirst: {
      marginTop: 0
    },
    areaHeader: {
      marginTop: 15
    },
    ratingArea: {
      display: 'flex'
    },
    typeSelect: {
      marginLeft: 10
    }
  })

interface Album {
  id: number
  tn_url: string
  name: string
  comment: string
  sub_categories: Album[]
}

interface Tag {
  id: number
  name: string
  counter: number
}

interface Column {
  label: string
  name: string
  direction: string
  enabled: boolean
}

export interface AlbumListItemProps extends WithStyles<typeof styles> {
  album: Album
  selectedAlbums: number[]
  onSelect: (albumID: number, selected: boolean) => void
}

function AlbumListItem(props: AlbumListItemProps) {
  const [foldersOpen, setFoldersOpen] = useState(false)

  const setOpen = (foldersOpen = false, e: MouseEvent) => {
    e.stopPropagation()
    setFoldersOpen(foldersOpen)
  }

  const { album, selectedAlbums, onSelect, classes } = props
  const isSelected = selectedAlbums.includes(album.id)

  return (
    <React.Fragment>
      <ListItemButton
        key={album.id}
        selected={isSelected}
        onClick={onSelect.bind(null, album.id, !isSelected)}
      >
        <ListItemAvatar>
          <Avatar alt={album.name} src={album.tn_url} />
        </ListItemAvatar>
        <ListItemText primary={album.name} secondary={album.comment} />
        {album.sub_categories?.length && !foldersOpen && (
          <Tooltip disableInteractive title="Show Sub-Albums">
            <ExpandMoreIcon onClick={setOpen.bind(this, true)} />
          </Tooltip>
        )}
        {album.sub_categories?.length && foldersOpen && (
          <Tooltip disableInteractive title="Hide Sub-Albums">
            <ExpandLessIcon onClick={setOpen.bind(this, false)} />
          </Tooltip>
        )}
      </ListItemButton>
      {foldersOpen && (
        <List className={classes.subList}>
          {album.sub_categories.map((album: Album) => (
            <AlbumListItem
              classes={classes}
              key={album.id}
              album={album}
              selectedAlbums={selectedAlbums}
              onSelect={onSelect}
            />
          ))}
        </List>
      )}
    </React.Fragment>
  )
}

export interface PiwigoDialogProps extends WithStyles<typeof styles> {
  open: boolean
  onClose: () => void
  onImportURL: (type: string, e: MouseEvent, ...args: any[]) => void
}

function PiwigoDialog(props: PiwigoDialogProps) {
  const { pathSep } = useAppSelector(selectConstants())
  const piwigoPassword = useAppSelector(
    selectAppConfigRemoteSettingsPiwigoPassword()
  )
  const piwigoUsername = useAppSelector(
    selectAppConfigRemoteSettingsPiwigoUsername()
  )
  const piwigoHost = useAppSelector(selectAppConfigRemoteSettingsPiwigoHost())
  const piwigoProtocol = useAppSelector(
    selectAppConfigRemoteSettingsPiwigoProtocol()
  )

  const [listType, setListType] = useState(PW.apiTypeCategory)
  const [albums, setAlbums] = useState<Album[]>([])
  const [tags, setTags] = useState<Tag[]>([])
  const [loggedIn, setLoggedIn] = useState(false)
  const [selectedAlbums, setSelectedAlbums] = useState<number[]>([])
  const [selectedTags, setSelectedTags] = useState<number[]>([])
  const [tagModeAnd, setTagModeAnd] = useState(false)
  const [sortRandom, setSortRandom] = useState(false)
  const [recursiveMode, setRecursiveMode] = useState(false)
  const [sortOrder, setSortOrder] = useState(
    Object.values(PWS)
      .filter((pw) => pw !== PWS.sortOptionRandom)
      .map((pw) => {
        return {
          label: en.get(pw),
          name: pw,
          direction: 'DESC',
          enabled: false
        } as Column
      })
  )

  const onDialogEntered = () => {
    if (listType === PW.apiTypeCategory) {
      getAlbums()
    } else if (listType === PW.apiTypeTag) {
      getTags()
    }
  }

  const onListTypeChange = (e: MouseEvent) => {
    const type = (e.target as HTMLInputElement).value
    setListType(type)
    if (type === PW.apiTypeCategory) {
      getAlbums()
    } else if (type === PW.apiTypeTag) {
      getTags()
    }
  }

  const login = async () => {
    await wretch(makeURL())
      .addon(AbortAddon())
      .addon(FormUrlAddon)
      .formUrl({
        method: 'pwg.session.login',
        username: piwigoUsername,
        password: piwigoPassword
      })
      .post()
      .setTimeout(5000)
      // .notFound((e) => pm({
      //   error: e.message,
      //   helpers: helpers,
      //   source: source,
      //   timeout: timeout,
      //   pathSep: pathSep
      // }))
      // .internalError((e) => pm({
      //   error: e.message,
      //   helpers: helpers,
      //   source: source,
      //   timeout: timeout,
      //   pathSep: pathSep
      // }))
      .json((json) => {
        if (json.stat === 'ok') {
          setLoggedIn(true)
        } else {
          //
        }
      })
      .catch((e) => {
        //
      })
  }

  const getAlbums = () => {
    const getAlbums = async () => {
      await wretch(makeURL())
        .addon(AbortAddon())
        .addon(FormUrlAddon)
        .formUrl({
          method: 'pwg.categories.getList',
          recursive: true,
          tree_output: true
        })
        .post()
        .setTimeout(5000)
        // .notFound((e) => pm({
        //   error: e.message,
        //   helpers: helpers,
        //   source: source,
        //   timeout: timeout,
        //   pathSep: pathSep
        // }))
        // .internalError((e) => pm({
        //   error: e.message,
        //   helpers: helpers,
        //   source: source,
        //   timeout: timeout,
        //   pathSep: pathSep
        // }))
        .json((json) => {
          if (json.stat === 'ok') {
            setAlbums(json.result.map((a: Album) => a))
          } else {
            //
          }
        })
        .catch((e) => {
          //
        })
    }

    if (!loggedIn && !!piwigoUsername) {
      login().then(getAlbums)
    } else {
      getAlbums()
    }
  }

  const getTags = () => {
    const getTags = async () => {
      await wretch(makeURL())
        .addon(AbortAddon())
        .addon(FormUrlAddon)
        .formUrl({ method: 'pwg.tags.getList' })
        .post()
        .setTimeout(5000)
        // .notFound((e) => pm({
        //   error: e.message,
        //   helpers: helpers,
        //   source: source,
        //   timeout: timeout,
        //   pathSep: pathSep
        // }))
        // .internalError((e) => pm({
        //   error: e.message,
        //   helpers: helpers,
        //   source: source,
        //   timeout: timeout,
        //   pathSep: pathSep
        // }))
        .json((json) => {
          if (json.stat === 'ok') {
            setTags(json.result.tags.map((t: Tag) => t))
          } else {
            //
          }
        })
        .catch((e) => {
          //
        })
    }

    if (!loggedIn && !!piwigoUsername) {
      login().then(getTags)
    } else {
      getTags()
    }
  }

  const createAPICall = (e: MouseEvent) => {
    let url = `${makeURL()}&method=${listType}`

    if (listType === PW.apiTypeCategory) {
      url += '&' + selectedAlbums.map((catID) => `cat_id[]=${catID}`).join('&')
      if (recursiveMode) {
        url += '&recursive=true'
      }
    } else if (listType === PW.apiTypeTag) {
      url += '&' + selectedTags.map((tagID) => `tag_id[]=${tagID}`).join('&')
      if (tagModeAnd) {
        url += '&tag_mode_and=true'
      }
    }

    if (sortRandom) {
      url += '&order=random'
    } else {
      const sortLines = sortOrder
        .filter((col) => col.enabled)
        .map((col) => `${col.name} ${col.direction}`)
        .join(',')
      if (sortLines) {
        url += `&order=${encodeURIComponent(sortLines)}`
      }
    }

    props.onImportURL(AF.url, null, [url])
  }

  const makeURL = () => {
    return piwigoProtocol + '://' + piwigoHost + '/ws.php?format=json'
  }

  const addSelectedAlbum = (albumID: number) => {
    setSelectedAlbums([...selectedAlbums, albumID])
  }

  const removeSelectedAlbum = (albumID: number) => {
    setSelectedAlbums(selectedAlbums.filter((t: number) => t !== albumID))
  }

  const addSelectedTag = (tagID: number) => {
    setSelectedTags([...selectedTags, tagID])
  }

  const removeSelectedTag = (tagID: number) => {
    setSelectedTags(selectedTags.filter((t: number) => t !== tagID))
  }

  const setRandomSortOrder = (sortRandom: boolean) => {
    setSortRandom(sortRandom)
  }

  const setRecursive = (recursiveMode: boolean) => {
    setRecursiveMode(recursiveMode)
  }

  const setTagMode = (tagModeAnd: boolean) => {
    setTagModeAnd(tagModeAnd)
  }

  const toggleSortColumn = (columnName: string, enabled: boolean) => {
    const newSortOrder = Array.from(sortOrder).map((col: Column) => {
      if (col.name === columnName) {
        return { ...col, enabled }
      }
      return col
    })
    setSortOrder(newSortOrder)
  }

  const setColumnDirection = (columnName: string, direction: string) => {
    const newSortOrder = Array.from(sortOrder).map((col: Column) => {
      if (col.name === columnName) {
        return { ...col, enabled: true, direction }
      }
      return col
    })
    setSortOrder(newSortOrder)
  }

  const classes = props.classes

  return (
    <Dialog
      open={props.open}
      onClose={props.onClose.bind(this)}
      TransitionProps={{ onEntered: onDialogEntered.bind(this) }}
      fullWidth={true}
      aria-labelledby="url-import-title"
      aria-describedby="url-import-description"
    >
      <DialogTitle id="url-import-title">
        Create a New Piwigo Source
      </DialogTitle>
      <DialogContent>
        <FormControl variant="standard">
          <Typography
            component="h2"
            variant="h6"
            className={classes.areaHeaderFirst}
          >
            Piwigo Source Type
          </Typography>
          <DialogContentText>
            Select the type of image list to create
          </DialogContentText>
          <Select
            variant="standard"
            value={listType}
            className={classes.typeSelect}
            onChange={onListTypeChange.bind(this)}
          >
            <MenuItem value={PW.apiTypeCategory}>Album Media</MenuItem>
            <MenuItem value={PW.apiTypeTag}>Tagged Media</MenuItem>
            <MenuItem disabled={!piwigoUsername} value={PW.apiTypeFavorites}>
              Your Favorites
            </MenuItem>
          </Select>
        </FormControl>
        {listType === PW.apiTypeCategory && (
          <React.Fragment>
            <Typography
              component="h2"
              variant="h6"
              className={classes.areaHeader}
            >
              Select an Album
            </Typography>
            <DialogContentText>
              Select the album to load media from
            </DialogContentText>
            <List className={classes.list}>
              {albums.map((album: Album) => (
                <AlbumListItem
                  classes={classes}
                  key={album.id}
                  album={album}
                  selectedAlbums={selectedAlbums}
                  onSelect={(albumID, isSelected) =>
                    this[
                      isSelected ? 'addSelectedAlbum' : 'removeSelectedAlbum'
                    ](albumID)
                  }
                />
              ))}
            </List>
            <FormControlLabel
              control={
                <Checkbox
                  onChange={setRecursive.bind(this, !recursiveMode)}
                  checked={recursiveMode}
                />
              }
              label="Recursive"
            />
          </React.Fragment>
        )}
        {listType === PW.apiTypeTag && (
          <React.Fragment>
            <Typography
              component="h2"
              variant="h6"
              className={classes.areaHeader}
            >
              Select Tags
            </Typography>
            <DialogContentText>
              Select the tags used to retrieve media with
            </DialogContentText>
            {tags.map((tag: Tag) => {
              const isSelected = selectedTags.includes(tag.id)
              return (
                <Chip
                  className={classes.tagChip}
                  key={tag.name}
                  color={isSelected ? 'secondary' : 'primary'}
                  onClick={
                    isSelected
                      ? removeSelectedTag.bind(this, tag.id)
                      : addSelectedTag.bind(this, tag.id)
                  }
                  label={tag.name}
                />
              )
            })}
            <Divider orientation="horizontal" flexItem />
            <FormControlLabel
              control={
                <Checkbox
                  onChange={setTagMode.bind(this, !tagModeAnd)}
                  checked={tagModeAnd}
                />
              }
              label="Must Match All Tags"
            />
          </React.Fragment>
        )}
        <Divider orientation="horizontal" flexItem />
        <React.Fragment>
          <Typography
            component="h2"
            variant="h6"
            className={classes.areaHeader}
          >
            Sort Order
          </Typography>
          <DialogContentText>
            Indicate the media sort order (optional)
          </DialogContentText>
          <FormControlLabel
            control={
              <Checkbox
                onChange={setRandomSortOrder.bind(this, !sortRandom)}
                checked={sortRandom}
              />
            }
            label="Randomize"
          />
          <Sortable
            className={sortRandom ? classes.sortListDisabled : null}
            options={{
              animation: 150,
              easing: 'cubic-bezier(1, 0, 0, 1)'
            }}
            onChange={(order: any, sortable: any, evt: any) => {
              const newSortOrder = Array.from(sortOrder)
              arrayMove(newSortOrder, evt.oldIndex, evt.newIndex)
              setSortOrder(newSortOrder)
            }}
          >
            {sortOrder.map((column) => (
              <Card className={classes.sortCol} key={column.name}>
                <CardContent
                  className={classes.sortColContent}
                  onClick={setColumnDirection.bind(
                    this,
                    column.name,
                    column.direction === 'ASC' ? 'DESC' : 'ASC'
                  )}
                >
                  <Checkbox
                    checked={column.enabled}
                    onChange={toggleSortColumn.bind(
                      this,
                      column.name,
                      !column.enabled
                    )}
                  />
                  <Typography
                    component="h3"
                    variant="h6"
                    className={classes.sortColLabel}
                  >
                    {column.label}
                  </Typography>
                  {column.direction === 'ASC' && (
                    <Tooltip disableInteractive title="Sort Ascending">
                      <ArrowUpwardIcon className={classes.sortColDir} />
                    </Tooltip>
                  )}
                  {column.direction === 'DESC' && (
                    <Tooltip disableInteractive title="Sort Descending">
                      <ArrowDownwardIcon className={classes.sortColDir} />
                    </Tooltip>
                  )}
                </CardContent>
              </Card>
            ))}
          </Sortable>
        </React.Fragment>
        {(listType === PW.apiTypeTag || listType === PW.apiTypeCategory) && (
          <React.Fragment>
            <Typography
              component="h2"
              variant="h6"
              className={classes.areaHeader}
            >
              Rating
            </Typography>
            <DialogContentText>
              Indicate the media sort order (optional)
            </DialogContentText>
            <Container className={classes.ratingArea}>
              <Container>
                <Typography component="legend">Minimum Rating</Typography>
                <Rating name="pwg-image-min" precision={0.5} />
              </Container>
              <Container>
                <Typography component="legend">Maximum Rating</Typography>
                <Rating name="pwg-image-max" precision={0.5} />
              </Container>
            </Container>
          </React.Fragment>
        )}
      </DialogContent>
      <DialogActions>
        <Button onClick={props.onClose.bind(this)} color="secondary">
          Cancel
        </Button>
        <Button
          // disabled={!importURL.match("^https?://") || (importType === GT.local && rootDir.length === 0)}
          onClick={createAPICall.bind(this)}
          color="primary"
        >
          Create List
        </Button>
      </DialogActions>
    </Dialog>
  )
}

;(PiwigoDialog as any).displayName = 'PiwigoDialog'
export default withStyles(styles)(PiwigoDialog as any)
