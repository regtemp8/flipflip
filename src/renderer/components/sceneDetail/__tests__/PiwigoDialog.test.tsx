import React from "react";
import { describe, it, expect } from "@jest/globals";
import renderer from "react-test-renderer";
import PiwigoDialog from "../PiwigoDialog";
import TestProvider from "../../../../../test/util/TestProvider";
import store from "../../../../store/store";

describe("PiwigoDialog", () => {
  it("should match snapshot", () => {
    const component = renderer.create(
      <TestProvider store={store}>
        <PiwigoDialog
          open={false}
          onClose={() => {}}
          onImportURL={(type: string, e: MouseEvent, ...args: any[]) => {}}
        />
      </TestProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
