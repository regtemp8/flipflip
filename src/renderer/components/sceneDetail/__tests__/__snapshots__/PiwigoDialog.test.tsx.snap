// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PiwigoDialog should match snapshot 1`] = `null`;
